### PR TITLE
SISRP-26984 - Pull Student Information from CS Endpoint for Transcript Link

### PR DESCRIPTION
**Pull request by davidlimcheng**
https://jira.berkeley.edu/browse/SISRP-26984

* Once this data is pulled, we will compose a POST request using this data that will be sent to Credential Solutions.  Once this request is processed, the student will have the form pre-filled and will not need to re-authorize.

### CODE REVIEW

File: fixtures/xml/campus_solutions/cs_official_transcript.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <SNM>Mike</SNM>
+  <SNL>Johnny</SNL>
+  <SNLO/>
+  <A1>79430 Berkeley Way</A1>
+  <A2/>
+  <CITY>Spanish Fort</CITY>
+  <STATE>AL</STATE>
+  <ZIP>36527</ZIP>
+  <CNTRY>USA</CNTRY>
+  <TEL25>205-222-6988</TEL25>
```
Comment (johncrossman) [inline fixtures/xml/campus_solutions/cs_official_transcript.xml]: @davidlimcheng I recommend you throw in dummy data for the line above.

File: app/models/campus_solutions/cs_official_transcript.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module CampusSolutions
+  class CsOfficialTranscript < CachedProxy
+
+    include CampusSolutionsIdRequired
+
+    def url
+      "#{@settings.base_url}/UC_SR_TRNSCPT_DATA.v1/Get?EMPLID=#{@campus_solutions_id}"
```
Comment (johncrossman) [inline app/models/campus_solutions/cs_official_transcript.rb]: @davidlimcheng would you mind adding this to `script/sis/verify_sis_api_endpoints.sh`.  You'd append the following to line 219:
```
  \
  "/UC_SR_TRNSCPT_DATA.v1/Get?EMPLID=${CAMPUS_SOLUTIONS_ID}"
```

**Comment by johncrossman** [comment]
Looks good, @davidlimcheng  I made two requests.

File: fixtures/xml/campus_solutions/cs_official_transcript.xml (hunk 1 above)
Comment (davidlimcheng) [inline fixtures/xml/campus_solutions/cs_official_transcript.xml]: Ah, good catch.  Will do.

**Comment by davidlimcheng** [comment]
Thanks @johncrossman, should be in tip-top shape now.

File: script/sis/verify_sis_api_endpoints.sh [hunk 3]
```diff
@@ -216,7 +216,8 @@ verify_cs 'always_enabled' true \
   "/UC_LINK_API.v1/get?PROPNAME=CALCENTRAL" \
   "/UC_AA_ADV_APPMT.v1/?EMPLID=${CAMPUS_SOLUTIONS_ID}" \
   "/UC_AA_ADV_STDNT_REL.v1/?EMPLID=${CAMPUS_SOLUTIONS_ID}" \
-  "/UC_AA_ADV_ACTION_ITEM.v1/?EMPLID=${CAMPUS_SOLUTIONS_ID}&CALLER_TYPE=ADVISOR"
+  "/UC_AA_ADV_ACTION_ITEM.v1/?EMPLID=${CAMPUS_SOLUTIONS_ID}&CALLER_TYPE=ADVISOR" \
+  "/UC_SR_TRNSCPT_DATA.v1/Get?EMPLID=${CAMPUS_SOLUTIONS_ID}"
```
Comment (johncrossman) [inline script/sis/verify_sis_api_endpoints.sh]: @davidlimcheng Thank you!

**Pull request merged**